### PR TITLE
Add `autonomy.test_tools`

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

TODO

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

n/a

### CODE REVIEW

File: autonomy/test_tools/configurations.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 # ------------------------------------------------------------------------------
 #
-#   Copyright 2021-2022 Valory AG
```
Comment (DavidMinarsch) [inline autonomy/test_tools/configurations.py:4]: keep range

File: tests/test_connections/test_abci/test_abci.py [hunk 2]
```diff
@@ -70,14 +74,11 @@
 from packages.valory.protocols.abci.dialogues import AbciDialogues as BaseAbciDialogues
 
 from tests.conftest import ANY_ADDRESS
```
Comment (DavidMinarsch) [inline tests/test_connections/test_abci/test_abci.py:76]: extract this too?

File: tests/test_connections/test_abci/test_abci.py [hunk 3]
```diff
@@ -70,14 +74,11 @@
...
-from tests.fixture_helpers import UseTendermint
 from tests.helpers.async_utils import (
```
Comment (DavidMinarsch) [inline tests/test_connections/test_abci/test_abci.py:77]: Extract?

File: tests/test_agents/test_registration.py [hunk 4]
```diff
@@ -25,11 +25,12 @@
...
 from aea.configurations.data_types import PublicId
 
+from autonomy.test_tools.fixture_helpers import UseACNNode, UseGnosisSafeHardHatNet
+
 from packages.valory.skills.registration_abci.behaviours import (
     RegistrationStartupBehaviour,
 )
 
-from tests.fixture_helpers import UseACNNode, UseGnosisSafeHardHatNet
 from tests.test_agents.base import BaseTestEnd2End, BaseTestEnd2EndExecution
```
Comment (DavidMinarsch) [inline tests/test_agents/test_registration.py:34]: Extract?

File: tests/test_agents/test_oracle_abci.py [hunk 5]
```diff
@@ -24,7 +24,8 @@
 import pytest
 from aea.configurations.data_types import PublicId
 
-from tests.fixture_helpers import UseGnosisSafeHardHatNet
+from autonomy.test_tools.fixture_helpers import UseGnosisSafeHardHatNet
+
 from tests.test_agents.base import BaseTestEnd2EndExecution, RoundChecks
```
Comment (DavidMinarsch) [inline tests/test_agents/test_oracle_abci.py:29]: Extract?

File: tests/test_agents/test_counter.py [hunk 6]
```diff
@@ -31,9 +31,10 @@
 import requests
 from aea.test_tools.test_cases import AEATestCaseMany
 
+from autonomy.test_tools.configurations import DEFAULT_REQUESTS_TIMEOUT
+from autonomy.test_tools.fixture_helpers import UseTendermint
+
 from tests.conftest import ANY_ADDRESS
-from tests.fixture_helpers import UseTendermint
-from tests.helpers.constants import DEFAULT_REQUESTS_TIMEOUT
 from tests.helpers.tendermint_utils import (
```
Comment (DavidMinarsch) [inline tests/test_agents/test_counter.py:38]: Extract?

File: tests/test_agents/test_apy_estimation_abci.py [hunk 7]
```diff
@@ -22,7 +22,8 @@
 import pytest
 from aea.configurations.data_types import PublicId
 
-from tests.fixture_helpers import UseGnosisSafeHardHatNet
+from autonomy.test_tools.fixture_helpers import UseGnosisSafeHardHatNet
+
 from tests.test_agents.base import BaseTestEnd2EndExecution, RoundChecks
```
Comment (DavidMinarsch) [inline tests/test_agents/test_apy_estimation_abci.py:27]: Extract?

File: tests/helpers/base.py [hunk 8]
```diff
@@ -19,18 +19,12 @@
 
 """Base helpers."""
```
Comment (DavidMinarsch) [inline tests/helpers/base.py:20]: Move whole file into autonomy?

File: setup.py [hunk 9]
```diff
@@ -52,7 +52,7 @@ def get_all_extras() -> Dict:
     "pandas<1.4,>=1.3.4",
     "watchdog >=2.1.6",
     "pytest==7.0.0",
-    "docker==5.0.3"
```
Comment (DavidMinarsch) [inline setup.py]: Can we maybe instead have this approach for Windows only and otherwise use the pypi distribution?

Comment (marcofavorito) [reply setup.py]: Good point, we can do it quite easily in `setup.py`; less easily in every testenv in `tox.ini` (but note that we install `docker` via `setup.py` in tox test environments; the ones we care for CI)

File: tox.ini [hunk 10]
```diff
@@ -66,7 +66,6 @@ basepython = python3.10
 usedevelop = True
 deps =
     {[testenv]deps}
-    git+ssh://github.com/docker/docker-py@3f0095a7c1966c521652314e524ff362c24ff58c#egg=docker
```
Comment (marcofavorito) [inline tox.ini]: here and below we don't need to install docker as it is already included in `setup.py`

File: setup.py (hunk 9 above)
Comment (marcofavorito) [reply setup.py]: [9ea2226](https://github.com/valory-xyz/open-autonomy/pull/1144/commits/9ea222662fb6741cabdd74fac5d2794e3b0b0ba1)

**Review by DavidMinarsch — APPROVED** [review]
LGTM -  Considering this merged ready as all the test failures are also on main.

**Pull request merged**